### ISSUE TITLE
Igame launches last game

### ISSUE THREAD

**Issue by prosonic69** [MrZammler/iGame#241]
igame will re-launch the last run game after quitting the game, if you are fast enough to quit igame before it relaunches the game and then load Igame a second time it will complain you need to make a selection 1st

pistorm classic with Pi3 or Pi4, amiga 2000 2mb chip ram, OS 3.2.1, latest release of Igame (done it with 2,1 also)

**Comment by walkero-gr** [comment]
@prosonic69 thank you for your patience. I remember some other guy had the same issue and that proved to be one of system patches. The OS that you use is based on a specific distribution? What version of MUI do you use?

**Comment by prosonic69** [comment]
@walkero-gr the OS is not a prebuilt one, clean loaded 3.2 with 3.2.1 update, MUI 3.8,

I thought i found the fix when i ported the OS over to the A4000 below a week ago as IGame would crash on load with the 060 it turns out it was a bug with displaying the game art settings (tool types & prefs file settings were clashing) applied the same settings to the 2000 without change.  the 2000 didn't have this symptom as it would load and launch games just fine, just re-launhes them. 

Ive also deleted the mui igame prefs as part of my troubleshooting. snoopdos doesn't reveal anything out of the normal, the oddity is I'm using the same os build on other machines aa3000+ RTG zuluscsi, A4000 060 zz9000 zuluscsi, A1200 Pistorm Lite Pi3/4, only real difference is 16bit v's 32bit AGA, I've updated the modules to suit the 2000 from 3.2 as well, i toyed with the idea of maybe the pi4 is to fast as it benchmarks over 2000 mibs to rule this out i underclocked the pi to similar speeds as the 1200 with the pi3 made no change, changing to a pi3 in the 2000 and the symptom persists.  i also changed when igame writes its stats to drive thinking maybe due to the speed its getting hung up in some kind of loop but considering when its loaded as second time its trying to launch a game instantly rules this out

 ill test some more today running the os with no patchs (MCP IPrefs etc no RTG) I'm sure I've done this but cant recall exactly so i will do this again today
 
 I'm yet to try a prebuilt os like caffeine to see if it replicates this maybe the next step i shun from using prebuilt os as get tend to be full of junk,

**Comment by walkero-gr** [comment]
@prosonic69 Thank you for sharing all that info and what you tried out. iGame starts a game when it gets an event from MUI that the user clicked on an item in the list.

Have in mind that the tooltypes are not used any more in the latest versions of iGame. They were removed some time ago.

I recently got a PiStorm, but I haven't tested iGame on it. I will do and will let you know if there is a problem with it. I am actually doing some streams on Twitch.tv/walkerogr setting this up, and I will definitely try it this tomorrow.

If I will be able to replicate it, I will also find a solution.

**Comment by walkero-gr** [comment]
Thank you for your continuous help. Which version of MUI are you using?

**Comment by prosonic69** [comment]
3.8

**Comment by walkero-gr** [comment]
Let me do some tests myself and will report back. I will test both 5 and 3.8 versions of MUI, but it would be interesting if you could make an installation of MUI 5 and see if you can replicate that problem.

**Comment by prosonic69** [comment]
tested with MUI 5, issue is still present

**Comment by walkero-gr** [comment]
@prosonic69 I wanted to let you know that I tried to replicate the problem you had on my pistormed A1200 and I couldn't. I actually did it during my stream last Friday, which you can find at https://www.twitch.tv/videos/2148147990
At the 20:00 and later, I started installing iGame and tried that by running a few games, but unfortunately the behaviour you described was not there.
Can you please let me know if there are any patches you use started from the startup-sequence or user-startup?

**Comment by prosonic69** [comment]
The problem is not present on my 1200 either as per my post, litterly same workbench is used on my pistorm lite as in my 2000 it was flat file copied, same patches and startup sequence Ill create a backup and do a clean load of workbench 3.2.1 no rtg and test Sent from my iPhoneOn 21 May 2024, at 08:53, George Sokianos ***@***.***> wrote:﻿
@prosonic69 I wanted to let you know that I tried to replicate the problem you had on my pistormed A1200 and I couldn't. I actually did it during my stream last Friday, which you can find at https://www.twitch.tv/videos/2148147990
At the 20:00 and later, I started installing iGame and tried that by running a few games, but unfortunately the behaviour you described was not there.
Can you please let me know if there are any patches you use started from the startup-sequence or user-startup?

**Comment by prosonic69** [comment]
tested vanilla install of 3.2 with 3.2.1 update, latest whdload user, mui 3.8, MCC_textEditor 15.56 MCCguifx installed, both mapped and phisical rom

Amiga 2000 rev6, Pistorm2k slot adapter, PI4, Emu68 V1.0, 2mb Chip

issue persists

**Comment by walkero-gr** [comment]
That is so weird. Can you please make a video of it?
Also, have you tried older versions? I would be interested how the following versions are working
https://github.com/MrZammler/iGame/releases/tag/v2.1
https://github.com/MrZammler/iGame/releases/tag/v2.2.1

**Comment by walkero-gr** [comment]
Are you using Emu68 or Sashimi? What version?
Since this is happening only to your A2000 I wonder if the problem is with the hardware setup of PiStorm. Maybe wrong kickstart loaded?

**Comment by prosonic69** [comment]
Emu68 V1.0, correct kickstart loaded i have the 3.2.1 ROM and Maprom using either doesn't change it,
V2.1 and V2.2.1 do it as well but take longer to re-launch the game as the database is slower.
Video wise there isn't much to see, i game loads and functions as expected as per normal it launches the games fine, just on hotkey quit from whdload, it launches the last game launched after about 2 seconds, you can quit again and the loop continues, for the record whdload loads and quits fine outside of igame

**Comment by walkero-gr** [comment]
> for the record whdload loads and quits fine outside of igame

Yeah, that's logical because the problem is on MUI. It seems that for some reason it sends a new event of a list item clicked when it's not.

Can I please ask you to test the following as well
https://github.com/MrZammler/iGame/releases/tag/v2.0
https://github.com/MrZammler/iGame/releases/tag/2.0b5

Sorry for asking to do so many checks, but it would be interesting to know whether a specific version works fine or not.

**Comment by prosonic69** [comment]
tested both versions same result

**Comment by walkero-gr** [comment]
Do you have similar issues with other MUI applications that have lists? For example, IBrowse bookmarks?
I am afraid the problem has nothing to do with iGame. If that still happens with these old versions, which were used by so many users for so long, I would say there is something else in your system that makes MUI behave like that.

**Comment by prosonic69** [comment]
ibrowse30 bookmarks etc are fine, its just igame

**Comment by prosonic69** [comment]
further testing, with standard and long-hold firmware on the pistorm2k made no difference

**Comment by pcotter72** [comment]
this looks like HW issue. Like a pressed joystick fire button is Always active .
Try this:
https://aminet.net/package/driver/input/JoyPortTest.lha
make sure lowlevel.library is activated and check the Fire signal. If its always ON, try a different joystick.

**Comment by prosonic69** [comment]
see my 3rd reply, i have already tested with no joystick connected, tested with joystick testing software and no stuck buttons

**Comment by pcotter72** [comment]
oops! 😬 my bad! I should have read the complete thread.
But definitely something is sending a press signal .. keyboard? Mouse...

**Comment by radact2** [comment]
Me too!
This is happening on my A2000 with a Pistorm2K/Pi3B+
I'm using a Wico Red Stick with no autofire so should be no signal and with both the original A2000 keyboard & a PS2 keyboard with an adapter, same result.

Annoying, but a reboot fixes it.

**Comment by pcotter72** [comment]
Strange... it seems that only the Amiga 2000 is affected? Which revision are you guys using?
Also replacing lowlevel.library changes anything for you?
https://aminet.net/package/driver/input/NewLowLevel

**Comment by prosonic69** [comment]
Rev6 Amiga 2000
[https://aminet.net/package/driver/input/NewLowLevel](url)
the above fixed this for me awesome find

**Comment by pcotter72** [comment]
This does not actually fix anything. It was more of a proof of concept.
This NEW library needs to be configured so that programs like IGAME that use lowlevel.library can detect joystick inputs. Like up and down to browse the game list and fire to start the game.

I remember that some A2000 revisions had such a problem, but I do not remember the fix.  Different joysticks would give different results....
Anyway, if I remember anything else I will report it.

**Comment by pcotter72** [comment]
Its possible to make Igame not to use lowlevel.library?

**Comment by walkero-gr** [comment]
Thank you guys for continuing this discussion. Your findings are really useful.
If the problem is the lowlevel.library, there is no way for iGame right now to have that disabled. I could add a switch in the next version.

**Comment by prosonic69** [comment]
can you add the option to toggle the year/played columns off as well
